models/classes/user/implementation/UserSettingsImpl.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2018-2021 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+namespace oat\tao\model\user\implementation;
+
+use oat\tao\model\user\UserSettings;
+
+class UserSettingsImpl implements UserSettings
```

Comment:
We generally name the interface with the "Interface" suffix to avoid having to add "Impl" on every class (Which is a Java standard, but we generally not use in PHP). 

```suggestion
class UserSettings implements UserSettingsInterface
```

Reply:
I prefer the other way around (interfaces (aka contracts) with no suffix, implementation with it), but no strong opinion either way.

However, if we do "swap things" here (suffix the interface and not the implementation), then we'll end up with this one and the service interface as `UserSettingsInterface` and `UserSettingsServiceInterface` while also having other interfaces (those that were already here) named as `LockoutStorage` and `UserLocks` (with no `-Interface` suffix) on the same tao-core namespace.

That seems inconsistent for me (and also was a reason to not use the suffix myself in the first place), but I'm OK with that if it is for you both, just wanted to raise my concerns first so we can make an informed decision.

OTOH adding `getUserSettings()` into the existing `GenerisUserService` may be an option worth considering as well.

(Also, I think we should not consider renaming also existing interfaces from tao-core to use the `-Interface` suffix since that would probably be a breaking change).

@gabrielfs7 @shpran Thoughts? Shall we change it, keep as it is now, merge into `GenerisUserService`?

Reply:
Swapped suffixes (interfaces with `-Interface` suffix, implementations with no suffix) in e03bf0aa8a187ae6e4651ba30d0c82b5ab73c616.